source/common.c

```diff
@@ -66,6 +66,14 @@ struct aws_allocator *aws_default_allocator(void) {
 }
 
 void *aws_mem_acquire(struct aws_allocator *allocator, size_t size) {
+    AWS_PRECONDITION(allocator != NULL);
+    AWS_PRECONDITION(allocator->mem_acquire != NULL);
+
+    if (size == 0) {
```

Comment:
I think just turn this into an `AWS_FATAL_ASSERT` and call it a day. OR, if you really want to always do the check always, wrap this check in `AWS_UNLIKELY`.

Reply:
@ColdenCullen Is there a way to write a test to check that the assert is indeed raised? I'd like to have a test for the behaviour.

Reply:
Not that I know if, but that would be radical! We could add an on-assert-hook or something, and write an `ASSERT_ASSERTS()` macro that installs the hook, runs a block, and uninstalls the hook. That's a lot to ask, but it would be neato. @justinboswell you have more experience with shit like this, whatchu think?

Reply:
I made an issue and a draft PR for it. 
https://github.com/awslabs/aws-c-common/issues/471
https://github.com/awslabs/aws-c-common/pull/472